src/org/omegat/gui/main/IMainMenu.java

```diff
@@ -37,23 +38,70 @@
  * @author Alex Buloichik (alex73mail@gmail.com)
  * @author Didier Briel
  * @author Aaron Madlon-Kay
+ * @author Hiroshi Miura
  */
 public interface IMainMenu {
+    /**
+     * Get Machine Translation menu object.
+     *
+     * @return machine translation menu.
+     */
     JMenu getMachineTranslationMenu();
 
+    /**
+     * Get Options menu object.
+     *
+     * @return options menu.
+     */
     JMenu getOptionsMenu();
 
+    /**
+     * Get tools menu object.
+     * @return tools menu.
+     */
     JMenu getToolsMenu();
 
+    /**
+     * Get Glossary menu object.
+     * @return glossary menu.
+     */
     JMenu getGlossaryMenu();
 
+    /**
+     * Get Project menu object.
+     * @return project menu.
+     */
     JMenu getProjectMenu();
 
+    /**
+     * Get AutoCompletion JMenu object.
+     * @return auto completion menu.
+     */
     JMenu getAutoCompletionMenu();
 
+    /**
+     * Get Help menu's JMenu object.
+     * @return help menu.
+     */
     JMenu getHelpMenu();
 
+    /**
+     * Get menu item specified in MenuKey marker.
+     *
+     * @param marker a menu item key.
+     * @return JMenu object.
+     */
     JMenu getMenu(MenuExtender.MenuKey marker);
 
+    /**
+     * Enable/Disable menu item.
+     * When it is JCheckBoxMenuItem and JRadioButtonMenuItem, a specified
+     * item will be selected.
+     *
+     * @param name    a component name of menu item.
+     * @param enabled true when make it enabled, otherwise make it disabled.
+     */
+    void enableMenuItem(String name, boolean enabled);
```

Comment:
Now we have two type of menu manipulation interface.
1. `getMenu(MenuKey)` 
2. `enableMenuItem(String name, boolean)`

When we define `name` for all JMenu components, we can replace No.1 with `getMenu(String name)`

Reply:
We can implement menu updates in Action class for these menu items.
The object can detect changes through ProperyChangeListener interface.